leshan-core/src/main/java/org/eclipse/leshan/json/JsonArrayElement.java

```diff
@@ -0,0 +1,174 @@
+	private static final long serialVersionUID = -7863478129361403779L;
+
+	@SerializedName("n")
+	private String name;
+
+	@SerializedName("v")
+	private Number floatValue;
+
+	@SerializedName("bv")
+	private Boolean booleanValue;
+
+	@SerializedName("ov")
+	private String objectLinkValue;
+
+	@SerializedName("sv")
+	private String stringValue;
+
+	@SerializedName("t")
+	private Integer time;
+
+	public String getName() {
+		return name;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	public Integer getTime() {
+		return time;
+	}
+
+	public void setTime(Integer time) {
+		this.time = time;
+	}
+
+	public void setFloatValue(Number floatValue) {
+		this.floatValue = floatValue;
+	}
+
+	public void setBooleanValue(Boolean booleanValue) {
+		this.booleanValue = booleanValue;
+	}
+
+	public void setObjectLinkValue(String objectLinkValue) {
+		this.objectLinkValue = objectLinkValue;
+	}
+
+	public void setStringValue(String stringValue) {
+		this.stringValue = stringValue;
+	}
+
+	public Object getResourceValue() {
+		
+		if (booleanValue != null) {
+			return booleanValue;
+		}
+		if (floatValue != null) {
+			return floatValue;
+		}
+		if (objectLinkValue != null) {
+			return objectLinkValue;
+		}
+		if (stringValue != null) {
+			return stringValue;
+		}
+		return null;
+	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result
+				+ ((booleanValue == null) ? 0 : booleanValue.hashCode());
+		result = prime * result
+				+ ((floatValue == null) ? 0 : floatValue.hashCode());
+		result = prime * result + ((name == null) ? 0 : name.hashCode());
+		result = prime * result
+				+ ((objectLinkValue == null) ? 0 : objectLinkValue.hashCode());
+		result = prime * result
+				+ ((stringValue == null) ? 0 : stringValue.hashCode());
+		result = prime * result + time;
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		JsonArrayElement other = (JsonArrayElement) obj;
+		if (booleanValue == null) {
+			if (other.booleanValue != null)
+				return false;
+		} else if (!booleanValue.equals(other.booleanValue))
+			return false;
+		if (floatValue == null) {
+			if (other.floatValue != null)
+				return false;
+		} else if (!floatValue.equals(other.floatValue))
+			return false;
+		if (name == null) {
+			if (other.name != null)
+				return false;
+		} else if (!name.equals(other.name))
+			return false;
+		if (objectLinkValue == null) {
+			if (other.objectLinkValue != null)
+				return false;
+		} else if (!objectLinkValue.equals(other.objectLinkValue))
+			return false;
+		if (stringValue == null) {
+			if (other.stringValue != null)
+				return false;
+		} else if (!stringValue.equals(other.stringValue))
+			return false;
+		if (time != other.time)
+			return false;
+		return true;
+	}
+
+	@Override
+	public String toString() {
```

Comment:
@jvermillard  Yes I did read the guidelines, However toString has conditional statements in this case, I don't think is good to have an empty string printed in toString output?
The value could be float, String, Boolean, string or objectLink. Only one value will be displayed in the to Sting output. The time value also can be empty.
I have seen StringBuilder has been used  in Leshan Core (ResourceModel.java, ObjectModel.java,...) ..
Let me know what you think?

Thanks